src/pages/Game.tsx

```diff
@@ -42,28 +45,38 @@ function Game() {
     navigator.mediaDevices
       .getUserMedia({ video: { width: 300 }, audio: true })
       .then((stream) => {
-        videoRef.current.srcObject = stream;
-        videoRef.current.autoplay = true;
-        videoRef.current.muted = false;
-        setmystream(stream);
+        if (videoRef.current) {
+          videoRef.current.srcObject = stream;
+          videoRef.current.autoplay = true;
+          videoRef.current.muted = false;
+          setmystream(stream);
+        }
       })
       .catch((err) => {
-        console.log(err.message);
+        setMessage(err.message);
         setDisplayVid(false);
         setAudio(false);
       });
   }, []);
+  useEffect(() => {
+    socket.emit('join_room', {
+      room: id,
+    });
+    return () => {
+      socket.emit('leave_room', id);
+    };
+  }, [id]);
   function toggleVideo() {
     if (displayVid) {
       setDisplayVid(false);
```

Comment:
In order to consolidate the `if` and `else` blocks, what about handling `setDisplayVid` like this?
```suggestion
      setDisplayVid(!displayVid);
```